pkg/preemptiontoleration/preemption_toleration_test.go

```diff
@@ -0,0 +1,241 @@
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/scheduler-plugins/pkg/apis/config"
+	"sigs.k8s.io/scheduler-plugins/pkg/apis/config/v1beta1"
+	"sigs.k8s.io/yaml"
+
+	v1 "k8s.io/api/core/v1"
+	schedulingv1 "k8s.io/api/scheduling/v1"
+
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/cache"
+	st "k8s.io/kubernetes/pkg/scheduler/testing"
+)
+
+func TestGetCompletedPreemptionToleration(t *testing.T) {
+	tests := []struct {
+		name          string
+		priorityClass *schedulingv1.PriorityClass
+		want          *config.PreemptionToleration
+	}{
+		{
+			name:          "nil just returns nil",
+			priorityClass: makePriorityClass(t, samplePriority, nil),
+			want:          nil,
+		},
+		{
+			name:          "empty preemption toleration completes MinimumPreemptablePriority",
+			priorityClass: makePriorityClass(t, samplePriority, &v1beta1.PreemptionToleration{}),
+			want: &config.PreemptionToleration{
+				MinimumPreemptablePriority: pointer.Int32Ptr(samplePriority + 1),
+			},
+		},
+		{
+			name: "preemption toleration does not complete TolerationSeconds",
+			priorityClass: makePriorityClass(t, samplePriority, &v1beta1.PreemptionToleration{
+				MinimumPreemptablePriority: pointer.Int32Ptr(samplePriority + 10),
+			}),
+			want: &config.PreemptionToleration{
+				MinimumPreemptablePriority: pointer.Int32Ptr(samplePriority + 10),
+			},
+		},
+		{
+			name: "no empty fields does not complete any fields",
+			priorityClass: makePriorityClass(t, samplePriority, &v1beta1.PreemptionToleration{
+				MinimumPreemptablePriority: pointer.Int32Ptr(samplePriority + 10),
+				TolerationSeconds:          pointer.Int64Ptr(10),
+			}),
+			want: &config.PreemptionToleration{
+				MinimumPreemptablePriority: pointer.Int32Ptr(samplePriority + 10),
+				TolerationSeconds:          pointer.Int64Ptr(10),
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := getCompletedPreemptionToleration(*tt.priorityClass)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if diff := cmp.Diff(tt.want, got); diff != "" {
+				t.Errorf("Unexpected result (-want, +got): %s", diff)
+			}
+		})
+	}
+}
+
+var (
+	samplePriority    = int32(100)
+	priorityClassName = "dummy"
+	now               = time.Now()
+)
+
+type testCase struct {
+	name                         string
+	victimCandidatePriorityClass *schedulingv1.PriorityClass
+	victimCandidate              *v1.Pod
+	preemptor                    *v1.Pod
+	wantErr                      bool
+	want                         bool
+	errSubStr                    string
+}
+
+func TestCanToleratePreemptionWithoutTolerationSeconds(t *testing.T) {
+	minimumPreemptablePriority := samplePriority + 10
+	preemptionToleration := &v1beta1.PreemptionToleration{
+		// it can tolerate preemption by p < samplePriority
+		// (i.e. samplePriority+1 can preempt it)
+		MinimumPreemptablePriority: pointer.Int32Ptr(minimumPreemptablePriority),
+	}
+	for _, tt := range []testCase{
```

Comment:
How about adding the case that CanToleratePreemption returns true without toleration seconds?

Reply:
LGTM